xdsl/transforms/stencil_shape_minimize.py

```diff
@@ -50,7 +76,14 @@ class StencilShapeMinimize(ModulePass):
 
     name = "stencil-shape-minimize"
 
+    restrict: tuple[int, ...] | None
+
     def apply(self, ctx: MLContext, op: builtin.ModuleOp) -> None:
+        PatternRewriteWalker(InvalidateTemps(), apply_recursively=False).rewrite_module(
+            op
+        )
+        restrict_store_op = RestrictStoreOp(restrict=self.restrict)
+        PatternRewriteWalker(restrict_store_op).rewrite_module(op)
```

Comment:
```suggestion
        if restrict:
            PatternRewriteWalker(
                [
                    InvalidateTemps(),
                    RestrictStoreOp(restrict=self.restrict),
                ],
                apply_recursively=False,
            ).rewrite_module(op)
```
This might be able to work in one pass (not sure though if this needs recursive application). There might also be a `shape-inference` pass inside the conditional, since this invalidates temps.

Reply:
this code gives a few errors in the IDE:
1. "restrict" is not defined" -> I'll change it to self.restrict
2. 
```
Argument of type "list[InvalidateTemps | RestrictStoreOp]" cannot be assigned to parameter "pattern" of type "RewritePattern" in function "__init__"
  "list[InvalidateTemps | RestrictStoreOp]" is not assignable to "RewritePattern"Pylance[reportArgumentType](https://github.com/microsoft/pyright/blob/main/docs/configuration.md#reportArgumentType)
```
```
Argument of type "list[InvalidateTemps | RestrictStoreOp]" cannot be assigned to parameter "pattern" of type "RewritePattern" in function "__init__"
  "list[InvalidateTemps | RestrictStoreOp]" is not assignable to "RewritePattern"Pylance[reportArgumentType](https://github.com/microsoft/pyright/blob/main/docs/configuration.md#reportArgumentType)
```
-> looking at other examples, if I add in an enclosed GreedyRewritePatternApplier and remove the ```apply_recursively=False```, it seems to get rid of this error but I don't fully understand it - I'll add it into a new commit and remove if necessary

Reply:
Yes, apologies!